### ISSUE TITLE
v3 Marshal generates invalid YAML

### ISSUE THREAD

**Issue by daenney** [go-yaml/yaml#457]
`input.yaml`:

```yaml
clusters:
  cmd:
  - id: "hi"
    name: "hi there"
```

Load, unmarshal, marshal, write, load, unmarshal:
```golang
package main

import (
	"fmt"
	"io/ioutil"
	"gopkg.in/yaml.v3"
)

type input struct {
	Clusters clusters `yaml:"clusters"`
}

type clusters struct {
	Cmd []cmd `yaml:"cmd"`
}

type cmd struct {
	ID string `yaml:"id"`
	Name string `yaml:"name"`
}

func main() {
	data, err := ioutil.ReadFile("./input.yaml")
	if err != nil {
		panic("failed to load")
	}

	i := input{}
	err = yaml.Unmarshal(data, &i)
	if err != nil {
		panic(fmt.Sprintf("failed to unmarshal %s", err))
	}

	d, err := yaml.Marshal(&i)
	if err != nil {
		panic("failed to marshal")
	}
	err = ioutil.WriteFile("marshalled.yaml", d, 0644)
	if err != nil {
		panic("failed to write")
	}
	data, err = ioutil.ReadFile("./marshalled.yaml")
	if err != nil {
		panic("failed to load after unmarshal/marshal")
	}

	i2 := input{}
	err = yaml.Unmarshal(data, &i2)
	if err != nil {
		panic("failed to unmarshal after unmarshal/marshal")
	}
}
```

Result:
```
panic: failed to unmarshal after unmarshal/marshal

goroutine 1 [running]:
main.main()
        /home/daenney/Development/github.com/daenney/goyamlboom/main.go:50 +0x1fb
exit status 2
```

The generated `marshalled.yaml`:
```
clusters:
    cmd:
    - id: hi
        name: hi there
```

The extra indentation of `name` breaks go-yaml, and it shouldn't be there.

```sh
~/D/g/d/goyamlboom ❯❯❯ cat go.mod 
module github.com/daenney/goyamlboom

go 1.12

require gopkg.in/yaml.v3 v3.0.0-20190409140830-cdc409dda467

~/D/g/d/goyamlboom ❯❯❯ cat go.sum
gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405 h1:yhCVgyC4o1eVCa2tZl7eS0r+SDo693bJlVdllGtEeKM=
gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
gopkg.in/yaml.v3 v3.0.0-20190409140830-cdc409dda467 h1:w3VhdSYz2sIVz54Ta/eDCCfCQ4fQkDgRxMACggArIUw=
gopkg.in/yaml.v3 v3.0.0-20190409140830-cdc409dda467/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
```

**Comment by daenney** [comment]
v2 works correctly, i.e when it marshalls it doesn't add the extra indent that breaks it

**Comment by daenney** [comment]
I suppose this isn't so much "can't unmarshal its own output" but more "marshals to invalid YAML". The indentation is erroneous. A extremely permissive interpreter may turn it into this instead:

```yaml
clusters:
  cmd:
    -
      id: "hi\nname: hi there"
```

Trying to load the generated `marshalled.yaml` with PyYAML equally blows up:

```python
import yaml

with open("marshalled.yaml") as f:
    try:
        print(yaml.safe_load(f))
    except Error as e:
        print(e)
```

```python
Traceback (most recent call last):
  File "withpython.py", line 5, in <module>
    print(yaml.safe_load(f))
  File "/usr/lib/python3.7/site-packages/yaml/__init__.py", line 94, in safe_load
    return load(stream, SafeLoader)
  File "/usr/lib/python3.7/site-packages/yaml/__init__.py", line 72, in load
    return loader.get_single_data()
  File "/usr/lib/python3.7/site-packages/yaml/constructor.py", line 35, in get_single_data
    node = self.get_single_node()
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 82, in compose_node
    node = self.compose_sequence_node(anchor)
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 111, in compose_sequence_node
    node.value.append(self.compose_node(node, index))
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/usr/lib/python3.7/site-packages/yaml/composer.py", line 127, in compose_mapping_node
    while not self.check_event(MappingEndEvent):
  File "/usr/lib/python3.7/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/usr/lib/python3.7/site-packages/yaml/parser.py", line 428, in parse_block_mapping_key
    if self.check_token(KeyToken):
  File "/usr/lib/python3.7/site-packages/yaml/scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "/usr/lib/python3.7/site-packages/yaml/scanner.py", line 220, in fetch_more_tokens
    return self.fetch_value()
  File "/usr/lib/python3.7/site-packages/yaml/scanner.py", line 576, in fetch_value
    self.get_mark())
yaml.scanner.ScannerError: mapping values are not allowed here
  in "marshalled.yaml", line 4, column 13
```

**Comment by niemeyer** [comment]
Thanks for the report. This is a broken interaction between the new indent customization support and my attempt to improve the formatting in these circumstances. I'll get it fixed.

**Comment by niemeyer** [comment]
This should have been fixed by 55513ca, which also improves the formatting in related cases.

Please let me know if you you find any other issues.
